patches/server/0803-Implement-server-resource-pack-API.patch

```diff
@@ -0,0 +1,84 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: yusshu <yushu@unnamed.team>
+Date: Thu, 2 Sep 2021 22:23:35 -0500
+Subject: [PATCH] Implement server resource pack API
+
+
+diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+index 7b6c547e71230fbb3733f99a4597b3f5b51547b8..78716a209c34e1fbb2db901e5729db664eaf7c4f 100644
+--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
++++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+@@ -88,8 +88,12 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
+     private MinecraftServerGui gui;
+     @Nullable
+     private final TextFilterClient textFilterClient;
++    // Paper start
+     @Nullable
+-    private final Component resourcePackPrompt;
++    private Component resourcePackPrompt;
++    @Nullable
```

Comment:
why make this nullable? from what I see you only set it to a `boolean` value and if it is null you use the server properties value, so why not just read the server property upon construction?

Reply:
done